### DISCUSSION TITLE
ROI detection slows video acquisition

### DISCUSSION THREAD

**Discussion by XeniaGofman** [bonsai-rx/bonsai#1839] [Q&A]
Hi, I'm using bonsai to detect if a mouse's paw is in a specific location during locomotion; when conditions are met, I send input to Arduino. Arduino sends TTL to another device every ~200ms (which triggers an electrical stim). The setup works well for about 2 minutes before slowing the video acquisition. 

I tried to crop the image, but it didn't solve the issue. If I increase the threshold so nothing is detected in the ROI, the video will run smoothly without freezing or delays. 
I've attached the workflow. Any ideas on how to solve this would be highly appreciated!

Thanks! 
<img width="712" alt="Screenshot 2024-06-10 163120" src="https://github.com/bonsai-rx/bonsai/assets/79152831/601db7c8-e872-4183-8773-b56c5c29b7e5">

**Comment by bruno-f-cruz** [comment]
Please include the workflow. Thanks

**Reply by XeniaGofman** [reply]
![xg](https://github.com/bonsai-rx/bonsai/assets/79152831/99289445-fc9e-45e0-b8f4-2a34d474e70b)

**Reply by bruno-f-cruz** [reply]
Please share the actual `.bonsai` file as instructed in #864

**Comment by XeniaGofman** [comment]
[XG_config_and_paw2.zip](https://github.com/user-attachments/files/15779213/XG_config_and_paw2.zip)

**Reply by bruno-f-cruz** [reply]
If you remove all writer nodes (video and image) does it work?
